src/utils/vm_mount.cpp

```diff
@@ -105,12 +107,44 @@ mp::VMMount::VMMount(const std::string& sourcePath,
     if (errors.size())
         throw std::runtime_error(
             fmt::format("Mount cannot apply mapping with duplicate ids:{}", fmt::to_string(errors)));
+
+    resolve_source_path();
 }
 
 mp::VMMount::VMMount(const QJsonObject& json) : VMMount{parse_json(json)} // delegate on copy ctor
 {
 }
 
+void mp::VMMount::resolve_source_path()
+{
+    std::error_code err;
+    fs::path path{source_path};
+
+    auto status = MP_FILEOPS.symlink_status(path, err);
+
+    if (status.type() == fs::file_type::symlink)
```

Comment:
Perhaps just need to remove this?

Reply:
I'm not sure, the C++ docs for [`read_symlink`](https://en.cppreference.com/w/cpp/filesystem/read_symlink) implies that it will error unless that check passes. But worth a try

Reply:
Why do we need `read_symlink` though? Can't we just run every input path through `weakly_canonical`.

Reply:
You're right, that would work!